DashAISoftware/DashAI
# Use of run endpoint

# Summary

 
The "runs" endpoint is used to create new runs.

## Type of change



- Front end new feature

## Changes


- The file `api/run.ts` is created to contain the functions to get and create runs
-  The function to create runs is used in `experiments/newExperimentModal.jsx`
- Temporary functions are created, which will be modified once the compatibility issues with the backend endpoints are resolved.

## How to Test


1. Go to `DashAI/DashAI/front` and run `yarn build`
2. Go back to `DashAI/` and run `python -c "import DashAI;DashAI.run()"`
3. Create a new experiment (follow the steps in the video in the "Screenshots" section)
4. To verify that your runs were created go to `localhost:8000/api/v1/docs` and try the "Get Runs" endpoint.

## Screenshots


[run_endpoint.webm](https://github.com/DashAISoftware/DashAI/assets/53657198/a82cc685-7f63-4e76-b90f-8d2225110672)

## Notes


In order to make the application functional, some modifications have been made to allow POST request functions to send certain parameters in the URL:
- `createExperimentTemp` in `api/experiment.ts`
- `createRun` in `api/run.ts`

These modifications will be corrected once the compatibility issues with the backend endpoints are resolved.
